.github/workflows/ci-dependabot-fixup.yml

```diff
@@ -1,118 +0,0 @@
-  pull_request_target:
-    types: [opened, synchronize, reopened]
-    branches:
-      - main
-
-# Cancel in-progress jobs on same branch.
-concurrency:
-  group: ${{ github.workflow }}-${{ github.event.pull_request.head.ref }}
-  cancel-in-progress: true
-
-jobs:
-  check-dependabot:
-    runs-on: ubuntu-latest
-    steps:
-    - name: Checkout code
-      uses: actions/checkout@v4
-      with:
-        ref: ${{ github.event.pull_request.head.ref }}
-        fetch-depth: "0"
-        # We use a Personal Access Token (PAT) to checkout and later push the commit instead of
-        # the default $GITHUB_TOKEN. This is because events triggered by $GITHUB_TOKEN will not
-        # trigger new workflow runs, but we want to re-run the CI after pushing the updated commit.
-        # https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow
-        token: ${{ secrets.BOT_GITHUB_TOKEN }}
-
-    - name: Ensure Dependebot author and Go mod updates
-      id: check
-      run: |
-        # Ensure Dependabot author.
-        if [ "${{ github.event.pull_request.user.login }}" != "dependabot[bot]" ]
-        then
-            echo "This PR was not created by Dependabot. No further action is being taken."
-            echo "skip=true" >> $GITHUB_OUTPUT
-            exit 0;
-        fi
-
-        # Ensure only Dependabot commits.
-        git fetch --no-tags origin +refs/heads/${BASE_BRANCH}:refs/remotes/origin/${BASE_BRANCH}
-        if git log origin/${BASE_BRANCH}..HEAD --pretty=format:'%an' | grep -v '^dependabot\[bot\]$' | grep -q .
-        then
-            echo "This PR has commits not by Dependabot."
-            echo "skip=true" >> $GITHUB_OUTPUT
-            exit 0;
-        fi
-
-        # Ensure Go dependency updates.
-        if ! git diff --name-only origin/${BASE_BRANCH}..HEAD | grep -q 'go\.mod$'
-        then
-          echo "This PR does not update any Go dependencies."
-          echo "skip=true" >> $GITHUB_OUTPUT
-          exit 0;
-        fi
-
-        echo "All commits are by Dependabot and update Go modules."
-      env:
-          BASE_BRANCH: ${{ github.base_ref }}
-
-    - name: Set up Go
-      if: steps.check.outputs.skip != 'true'
-      uses: actions/setup-go@v5
-      with:
-        go-version: "1.22.x"
-
-    - name: Build gomod updater
-      if: steps.check.outputs.skip != 'true'
-      working-directory: tools/gomod-updater
-      run: go build
-
-    # Dependabot titles are:
-    #      <prefix>: bump <dependancy name> from <frmo_version> to <to_version> in <path>
-    # e.g. client-sdk/go: bump github.com/ethereum/go-ethereum from 1.12.1 to 1.13.3 in /client-sdk/go
-    # as long as the (configurable) <prefix> is without whitespace, the bellow parsing should work.
-    - name: Try extracting package name and version
-      if: steps.check.outputs.skip != 'true'
-      id: extract
-      run: |
-        title="${{ github.event.pull_request.title }}"
```

Comment:
This was unsafe - dependabot could had injected code with a title `aaaa"; echo "inject`
https://github.com/oasisprotocol/oasis-rosetta-gateway/pull/527

and we were using `pull_request_target` :scream:  https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
> When a workflow is triggered by the [pull_request_target](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) event, the GITHUB_TOKEN is granted read/write repository permission, even when it is triggered from a public fork.

and this doesn't apply because it's about **base ref** https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#responding-to-events
> For workflows initiated by Dependabot (github.actor == 'dependabot[bot]') using the pull_request_target event, if the base ref of the pull request was created by Dependabot (github.actor == 'dependabot[bot]'), the GITHUB_TOKEN will be read-only and secrets are not available.